src/main/org/codehaus/groovy/ast/VariableScope.java

```diff
@@ -178,4 +178,26 @@ public Object removeReferencedClassVariable(String name) {
     public Iterator<Variable> getReferencedClassVariablesIterator() {
         return getReferencedClassVariables().values().iterator();
     }
-}
\ No newline at end of file
+
+    /**
+     * Gets a map containing the variables declared in this scope.
+     * This map cannot be modified.
+     * @return a map containing the declared variable references
+     */
+    public Map<String, Variable> getDeclaredVariables() {
```

Comment:
Actually `Collections.EMPTY_MAP` is an immutable map. See: https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#EMPTY_MAP
This means the map will be immutable in both cases since `new HashMap<String, Integer>()` is not equal to the `Collections.EMPTY_MAP`. I've reset the last commit.